cmd/maya-apiserver/cstor-operator/spc/storagepool_create.go

```diff
@@ -163,27 +163,25 @@ func (pc *PoolCreateConfig) withDisks(casPool *apis.CasPool, spc *apis.StoragePo
 	if err != nil {
 		return nil, errors.Wrapf(err, "aborting storagepool create operation as no node qualified")
 	}
-	if len(nodeBDs.BlockDevices.Items) == 0 {
-		return nil, errors.New("aborting storagepool create operation as no block device was found")
+
+	claimedNodeBDs, err := pc.ClaimBlockDevice(nodeBDs, spc)
+	if claimedNodeBDs == nil || len(claimedNodeBDs.BlockDeviceList) == 0 {
```

Comment:
`claimedNodeBDs` should not be nil if there are any claimedBDs as per semantics.
check for `err` to be not nil rather than claimedNodeBDs and make `ClaimBlockDevice` return the partial list of claimed ones

Reply:
Done